### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by leppie**
Hi @arcfide 

I picked a few little issues when adding IronScheme support.

See inline comments for the reason changing.

Thanks

### CODE REVIEW

File: %3a133/vectors.sls [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
     ;;; * Mutators
     vector-swap! vector-fill! vector-reverse! vector-copy! vector-reverse-copy!
-    vector-reverse! vector-unfold! vector-unfold-right!
+    vector-unfold! vector-unfold-right!
```
Comment (leppie) [inline %3a133/vectors.sls]: `vector-reverse!` is exported twice. Not sure if spec allows, but my version of psyntax does not like it.

File: %3a145/assumptions.sls [hunk 2]
```diff
@@ -1,6 +1,6 @@
 (library (srfi :145 assumptions)
   (export assume)
-  (import (rnrs) (srfi :0) (srfi :23))
+  (import (except (rnrs) error) (srfi :0) (srfi :23))
```
Comment (leppie) [inline %3a145/assumptions.sls]: `error` is exported from both rnrs and SRFI 23.

File: tests/list-queues.sps [hunk 3]
```diff
@@ -84,14 +86,14 @@
...
+  (define xx (list-queue-unfold done? double add1 0))
+  (test-equal '(0 2 4 6) (list-queue-list xx))
+  (define yy (list-queue-unfold-right done? double add1 0))
+  (test-equal '(6 4 2 0) (list-queue-list yy))
+  (define xx0 (list-queue 8))
+  (define xx1 (list-queue-unfold done? double add1 0 xx0))
+  (test-equal '(0 2 4 6 8) (list-queue-list xx1))
+  (define yy0 (list-queue 8))
+  (define yy1 (list-queue-unfold-right done? double add1 0 yy0))
+  (test-equal '(8 6 4 2 0) (list-queue-list yy1))
```
Comment (leppie) [inline tests/list-queues.sps]: As all variables are defined on top level scope, duplicates are not allowed

File: tests/list-queues.sps [hunk 4]
```diff
@@ -2,6 +2,8 @@
 ; Using SRFI-64 test-group gave some trouble hence using test-begin/test-end.
 
 (import
+  (rnrs)
+  (srfi private define-values)
```
Comment (leppie) [inline tests/list-queues.sps]: IronScheme complained that `define` and `define-values` could not be found

File: tests/tables-test.sps [hunk 5]
```diff
@@ -241,7 +241,7 @@
       '(#f #f #t #t #t #t #f #f #f #f #f #f #f #f #f #f))
 
 (test (map hash-table-mutable? test-tables)
-      '(#t #t #t #t #t #t #t #t #t #t #t #t #t #t #t #f))
+      '(#t #f #t #t #t #t #t #t #t #t #t #t #t #t #t #f))
```
Comment (leppie) [inline tests/tables-test.sps]: The second hash table is created with `hash-table` using the default #f value for mutable, hence it is not mutable.

File: tests/tables-test.sps [hunk 6]
```diff
@@ -450,7 +450,7 @@
                   '(169 144 121 0 1 4 9 16 25 36 49 64 75 81)))
       '(13 12 11 0 1 2 3 4 5 -1 -1 8 -1 -1))
 
-(let ((ht-eg (hash-table number-comparator 1 1 4 2 9 3 16 4 25 5 64 8)))
+#;(let ((ht-eg (hash-table number-comparator 1 1 4 2 9 3 16 4 25 5 64 8)))
```
Comment (leppie) [inline tests/tables-test.sps]: The hash table is created with `hash-table` using the default #f value for mutable, hence it is not mutable.

File: tests/tables-test.sps [hunk 7]
```diff
@@ -630,11 +630,11 @@
 
 (test (begin (hash-table-union! ht-eqv2 ht-fixnum)
              (hash-table=? default-comparator ht-eqv2 ht-fixnum))
-      #t)
+      #f)
```
Comment (leppie) [inline tests/tables-test.sps]: `hash-table=?` requires both hashtables to have the same equivalence function, but they are `eqv?` and `=` respectively.

File: tests/tables-test.sps [hunk 8]
```diff
@@ -630,11 +630,11 @@
...
 
 (test (begin (hash-table-intersection! ht-eqv2 ht-fixnum)
              (hash-table=? default-comparator ht-eqv2 ht-fixnum))
-      #t)
+      #f)
```
Comment (leppie) [inline tests/tables-test.sps]: `hash-table=?` requires both hashtables to have the same equivalence function, but they are `eqv?` and `=` respectively.

**Comment by arcfide** [comment]
Good Fixes, thanks.

**Pull request merged**